### ISSUE TITLE
Cast error for `hugo new blabla/hello.md` with custom theme (due to empty archetype file)

### ISSUE THREAD

**Issue by Diti** [spf13/hugo#315]
Hello!

After creating a new theme and attempting to use Hugo's builder for content, I get the following error: `ERROR: 2014/06/01 Unable to Cast <nil> to map[string]interface{}`.

You can replicate the problem with Hugo v0.11 and the commands below.

    hugo new site test && cd test
    hugo new theme mytheme
    printf 'theme = "mytheme"\n' >> config.toml
    hugo new thispageworks.md
    hugo new this/fails.md

Am I doing anything wrong? I think this may be a bug.

**Comment by anthonyfok** [comment]
Hi @Diti,

Thank you for your bug report.  Actually, I think something could be done to improve the behaviour.
If I understand correctly, the same problem that @Diti experienced with v0.11 remains with us today with v0.14-DEV:

```
/tmp$ hugo new site test && cd test
/tmp/test$ hugo new theme mytheme
/tmp/test$ printf 'theme = "mytheme"\n' >> config.toml
/tmp/test$ hugo new thispageworks.md
ERROR: 2015/03/25 Error processing archetype file: /tmp/test/themes/mytheme/archetypes/default.md
ERROR: 2015/03/25 Unable to Cast <nil> to map[string]interface{}
/tmp/test$ hugo new this/fails.md
ERROR: 2015/03/25 Error processing archetype file: /tmp/test/themes/mytheme/archetypes/default.md
ERROR: 2015/03/25 Unable to Cast <nil> to map[string]interface{}
```

Reason?  `themes/mytheme/archetypes/default.md` is empty, and `hugo new` does not like that.
Ditto with a malformed `archetypes/default.md`.  For example, if I were to write just this line:

    author = "Anthony Fok"

in my `archetypes/default.md` file, without the expected surrounding `+++` for denoting TOML, `hugo new` would give me the same cryptic `ERROR: 2015/03/25 Unable to Cast <nil> to map[string]interface{}` error, which only a seasoned Go developer would understand.  And no, we cannot / should not expect our Hugo end users to be Go developers.

I think one or both of the following should be done:

 1. `hugo new theme` should probably generate a proper `archetypes/default.md` file, or just leave it out altogether.
 2. `hugo new [post]` should catch a malformed archetype file error, and should explicitly tell the user that the archetype file is malformed instead of showing that cryptic error.

**Comment by CMHarrington** [comment]
I'm getting the same error when copying/pasting the code from Step 1 here: http://gohugo.io/content/archetypes/ into a new document.

Saving to archetypes/default.md, then trying to create new post/newfrontmatter.md gives me the same error.  `ERROR: 2015/04/05 Unable to Cast <nil> to map[string]interface{}`

I also heeded the caveats, and added an extra line after the end +++ hoping that would work, but it doesn't. (I don't seem to be getting an EOF error, rather a cast error as above)

FWIW, I've tried this on a linux machine (emacs), and on a mac (using BBEdit).
